loadtest/store/memstore/store.go

```diff
@@ -775,21 +776,22 @@ func (s *MemStore) SetEmojis(emoji []*model.Emoji) error {
 	return nil
 }
 
-// SetReactions stores the given reactions for the specified post.
-func (s *MemStore) SetReactions(postId string, reactions []*model.Reaction) error {
-	s.lock.Lock()
-	defer s.lock.Unlock()
-
-	s.reactions[postId] = reactions
-	return nil
-}
-
 // SetReaction stores the given reaction.
 func (s *MemStore) SetReaction(reaction *model.Reaction) error {
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	s.reactions[reaction.PostId] = append(s.reactions[reaction.PostId], reaction)
+	// We get an element from the queue and check if we have it in the map and
+	// if it points to the same memory location. If so, we delete it since it means the queue is full.
+	// This is done to keep the data pointed by the map consistent with the data stored in the queue.
+	r := s.reactionsQueue.Get()
+	if rs, ok := s.reactions[r.PostId]; ok && rs[len(s.reactions[r.PostId])-1] == r {
+		rs[len(s.reactions[r.PostId])-1] = nil
+		s.reactions[r.PostId] = rs[:len(rs)-1]
+	}
+
+	*r = *reaction
+	s.reactions[r.PostId] = append(s.reactions[r.PostId], r)
```

Comment:
The basic idea is that Cqueue is a ring buffer of pointers and as we keep adding reactions, the buffer gets filled up. When it gets full, it means the next pointer would already exist in the map (since it's circular). So we delete the entry in the map, and overwrite the pointer value in the buffer to be the new one.

This way we keep the total number of entries in the map to be limited.